src/screens/WriteExerciseScreen.tsx

```diff
@@ -79,29 +78,14 @@ const WriteExerciseScreen = ({ navigation, route }: WriteExerciseScreenPropsType
   const { extraParams, retryData } = route.params
   const { trainingSet, trainingSetId, disciplineTitle } = extraParams
   const [isModalVisible, setIsModalVisible] = useState(false)
-  const [documents, setDocuments] = useState<DocumentsType>([])
   const [isLoading, setIsLoading] = useState(true)
-  const [currentDocumentNumber, setCurrentDocumentNumber] = useState(1)
-
-  useFocusEffect(
-    React.useCallback(() => {
-      setCurrentDocumentNumber(0)
-      const getDocuments = async (): Promise<void> => {
-        try {
-          const documentsRes =
-            retryData !== undefined
-              ? retryData
-              : await axios.get(ENDPOINTS.documents.all.replace(':id', `${trainingSetId}`))
-          setDocuments(documentsRes.data)
-          setCurrentDocumentNumber(0)
-        } catch (error) {
-          console.error(error)
-        }
-      }
-
-      getDocuments()
-    }, [retryData, trainingSetId])
-  )
+  const [currentDocumentNumber, setCurrentDocumentNumber] = useState(0)
+
+  let documents = useLoadDocuments(trainingSetId).data
+
+  if (retryData) {
+    documents = retryData.data
+  }
```

Comment:
```suggestion
  const documents = retryData?.data ?? useLoadDocuments(trainingSetId).data
```

Reply:
Yeah, wanted to do this as well, but it does not work as `ESLint: React Hook "useLoadDocuments" is called conditionally. React Hooks must be called in the exact same order in every component render.(react-hooks/rules-of-hooks`